File: dev/test_data/main/typescript/expected/implementation_specific/input/snippets/Verification/items_have_unique_labels.ts
/**
 * Check that {@link types.IItem.label}'s of the `items` do not repeat.
 *
 * @param items - to be verified
 * @returns `true` if the check passes
 */
export function itemsHaveUniqueLabels(
  items: Iterable<AasTypes.IItem>
): boolean {
  const labelSet = new Set<string>();
  for (const item of items) {
    if (labelSet.has(item.label)) {
      return false;
    }

    labelSet.add(item.label);
  }

  return true;
}

File: dev/test_data/main/python/expected/implementation_specific/input/snippets/Verification/items_have_unique_labels.py
def items_have_unique_labels(items: Iterable[aas_types.Item]) -> bool:
    """
    Check that :py:attr:`.types.Item.label`'s of the :paramref:`items`
    do not repeat.
    """
    label_set = set()  # type: Set[str]
    for item in items:
        if item.label in label_set:
            return False

        label_set.add(item.label)

    return True
